shared-bindings/time/__init__.c

```diff
@@ -83,12 +83,10 @@ mp_obj_t struct_time_make_new(const mp_obj_type_t *type, size_t n_args, const mp
     if (n_args != 1 || (kw_args != NULL && kw_args->used > 0)) {
         return namedtuple_make_new(type, n_args, args, kw_args);
     }
-    if (mp_obj_get_type(args[0])->getiter != mp_obj_tuple_getiter || ((mp_obj_tuple_t *)MP_OBJ_TO_PTR(args[0]))->len != 9) {
-        mp_raise_TypeError(translate("time.struct_time() takes a 9-sequence"));
-    }
-
-    mp_obj_tuple_t *tuple = MP_OBJ_TO_PTR(args[0]);
-    return namedtuple_make_new(type, 9, tuple->items, NULL);
+    size_t len;
+    mp_obj_t *items;
+    mp_obj_get_array(args[0], &len, &items);
+    return namedtuple_make_new(type, len, items, NULL);
```

Comment:
```suggestion
    if (len != 9) {
        mp_raise_TypeError(translate("time.struct_time() takes a 9-sequence"));
    }
    return namedtuple_make_new(type, len, items, NULL);
```

Reply:
The messages given by standard Python when this constructor is used wrongly are varied:
```
TypeError: structseq() takes at most 2 arguments (5 given)
TypeError: time.struct_time() takes an at least 9-sequence (5-sequence given)
TypeError: constructor requires a sequence
TypeError: time.struct_time() takes an at most 11-sequence (14-sequence given)
```

In this case, I think it's worth considering that taking away one of those messages doesn't actually hurt error checking (incorrect code is still rejected) when it saves some flash size for the additional check and the translated message.

Reply:
On further reflection, I've accepted your suggestion.  Thanks!